clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -166,20 +187,27 @@ public <P> Builder addPartitions(
             return this;
         }
 
-        public Builder merge(
-            OffsetCommitResponseData newData
-        ) {
+        public Builder<T> merge(OffsetCommitResponseData newData) {
+            if (version >= 9) {
+                // This method is called after the group coordinator committed the offsets. The group coordinator
+                // provides the OffsetCommitResponseData it built in the process. As of now, this data does
+                // not contain topic ids, so we resolve them here.
+                newData.topics().forEach(
+                        topic -> topic.setTopicId(topicResolver.getTopicId(topic.name()).orElse(Uuid.ZERO_UUID)));
```

Comment:
At this point, topic ids should be always resolvable. However if some aren't, we should fallback to adding the topic "as is" to the response to avoid caching `ZERO_UUID` with risk of overwrites.